MultipleFileUpload/MultipleFileUpload.php

```diff
@@ -63,37 +63,35 @@ class MultipleFileUpload extends UploadControl
 	 */
 	public static $baseWWWRoot = null;
 
+	/** @var Nette\Http\IRequest */
+	private static $request;
 
-    /** @var Nette\DI\Container */
-    private static $container;
+	/** @var bool */
+	private static $productionMode;
 
 	/**
 	 * Initialize MFU
 	 */
-	public static function init()
+	public static function init(Nette\Http\IRequest $request, Array $parameters)
```

Comment:
static methods from this class init(), register(), ... needs to be split into another class that will be created and registered in constructor. MFU will then take this object as a dependency.